IPython/html/widgets/widget_output.py

```diff
@@ -38,13 +38,30 @@ def clear_output(self, *pargs, **kwargs):
             clear_output(*pargs, **kwargs)
 
     def __enter__(self):
+        """Called upon entering output widget context manager."""
         self._flush()
-        self.send({'method': 'push'})
+        kernel = get_ipython().kernel
+        session = kernel.session
+        send = session.send
+        self._original_send = send
+        self._session = session
+
+        def send_hook(stream, msg_or_type, *args, **kwargs):            
+            if stream is kernel.iopub_socket and msg_or_type in ['clear_output', 'stream', 'display_data']:
```

Comment:
I don't think so, but I may not understand all of the different entry points to this method.  The time which I saw a full message being passed in was for the repr/display hook, which shouldn't be captured in the context anyways.